### PR TITLE
[pyright] [gql] AssetKey cleanup

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Fix some type errors by making it so that `AssetKey.from_graphql_input` always returns an `AssetKey` (previously it could also return `None`) and cleans up associated logic around callsites.

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11002" target="_blank">Graphite</a> 
    * **PR #11330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11330" target="_blank">Graphite</a> 
    * **PR #11001** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11001" target="_blank">Graphite</a> 
    * **PR #11038** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11038" target="_blank">Graphite</a> 
      * **PR #11051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11051" target="_blank">Graphite</a> 
      * **PR #11362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11362" target="_blank">Graphite</a> 
        * **PR #11717** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11717" target="_blank">Graphite</a> 
        * **PR #11695** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11695" target="_blank">Graphite</a> 
          * **PR #11696** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11696" target="_blank">Graphite</a> 
            * **PR #11697** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11697" target="_blank">Graphite</a> 
              * **PR #11698** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11698" target="_blank">Graphite</a> 
                * **PR #11699** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11699" target="_blank">Graphite</a> 
                  * **PR #11363** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11363" target="_blank">Graphite</a> 
                    * **PR #11364** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11364" target="_blank">Graphite</a> 
                      * **PR #11365** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11365" target="_blank">Graphite</a> 
                        * **PR #11366** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11366" target="_blank">Graphite</a> 
                          * **PR #11367** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11367" target="_blank">Graphite</a> 
                            * **PR #11368** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11368" target="_blank">Graphite</a> 
                              * **PR #11369** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11369" target="_blank">Graphite</a> 
                                * **PR #11720** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11720" target="_blank">Graphite</a> 
                                  * **PR #11721** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11721" target="_blank">Graphite</a>  👈
                                * **PR #10983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10983" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11721?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py [hunk 1]
```diff
@@ -665,7 +665,8 @@ def resolve_assetNodes(self, graphene_info, **kwargs):
         ]
 
     def resolve_assetNodeOrError(self, graphene_info, **kwargs):
-        return get_asset_node(graphene_info, AssetKey.from_graphql_input(kwargs["assetKey"]))
+        asset_key_input = cast(Mapping[str, Sequence[str]], kwargs.get("assetKey"))
```
Comment (schrockn) [inline python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:668]: Shouldn't we be doing a dynamic runtime check here to match the cast to ensure that the `cast` isn't lying?

**Review by schrockn — APPROVED** [review]
Have slight preference here for factoring out the cast to a helper function that also does a runtime check, converts `Sequence[AssetKey]` and notes that this is always paired with a graphene field with type `graphene.Argument(non_null_list(GrapheneAssetKeyInput))`

File: python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:668]: >Have slight preference here for factoring out the cast to a helper function that also does a runtime check, converts Sequence[AssetKey] and notes that this is always paired with a graphene field with type graphene.Argument(non_null_list(GrapheneAssetKeyInput))

This is addressed upstack in https://github.com/dagster-io/dagster/pull/11724

That PR eliminates all use of `**kwargs` in resolvers so no casts are needed anymore.

**Pull request merged**